### PR TITLE
Prepare data structures for adding snow

### PR DESCRIPTION
**Pull request by trontrytel**
This PR adds all the data structures for storing snow and its tendencies.  Right now all the source terms related to snow are still set to zero. So this PR should not bring any changes.

### CODE REVIEW

File: src/EDMF_Environment.jl [hunk 1]
```diff
@@ -1,11 +1,20 @@
-function update_env_precip_tendencies(en_thermo::EnvironmentThermodynamics, state, k, qt_tendency, θ_liq_ice_tendency)
-
+function update_env_precip_tendencies(
+    en_thermo::EnvironmentThermodynamics,
```
Comment (charleskawczynski) [inline src/EDMF_Environment.jl:2]: It looks like we can remove the `en_thermo` argument?

File: src/EDMF_Precipitation.jl [hunk 2]
```diff
@@ -60,22 +73,36 @@ function compute_rain_evap_tendencies(::PrecipPhysics, grid, state, gm, TS::Time
...
+        # When we fuse loops, this should hopefully disappear
         q_tot_gm = prog_gm.q_tot[k]
         T_gm = aux_gm.T[k]
-        # When we fuse loops, this should hopefully disappear
-        ts = TD.PhaseEquil_pTq(param_set, p0_c[k], T_gm, q_tot_gm)
+        p0 = p0_c[k]
+        ρ0 = ρ0_c[k]
+        ts = TD.PhaseEquil_pTq(param_set, p0, T_gm, q_tot_gm)
         q = TD.PhasePartition(ts)
+        qv = TD.vapor_specific_humidity(ts)
```
Comment (charleskawczynski) [inline src/EDMF_Precipitation.jl:86]: Is this used anywhere?

File: src/EDMF_Precipitation.jl [hunk 3]
```diff
@@ -60,22 +73,36 @@ function compute_rain_evap_tendencies(::PrecipPhysics, grid, state, gm, TS::Time
...
+        qr = prog_pr.q_rai[k]
+        qs = prog_pr.q_sno[k]
```
Comment (charleskawczynski) [inline src/EDMF_Precipitation.jl:88]: Is this used anywhere?

**Review by charleskawczynski — APPROVED** [review]
Overall LGTM!

File: src/EDMF_Precipitation.jl (hunk 3 above)
Comment (trontrytel) [reply src/EDMF_Precipitation.jl:88]: Will be used in the following PR with source terms computations.

File: src/EDMF_Environment.jl (hunk 1 above)
Comment (trontrytel) [reply src/EDMF_Environment.jl:2]: Thats true. I'll delete it. The corresponding function from updrafts is different. I don't think we will be dispatching over them.

**Comment by trontrytel** [comment]
Let's see the CI plots. I want to double check I'm not changing anything for RiCO

**Comment by charleskawczynski** [comment]
Oh, yeah, something is wrong with Rico

**Review by charleskawczynski — COMMENTED** [review]
I think we can decompose this PR into smaller pieces.

**Pull request closed without merging**